src/components/VideoDetail/VideoDetail.js

```diff
@@ -16,40 +18,64 @@ const VideoPlayer = styled.iframe`
+            return favorites.findIndex(item => JSON.stringify(item) === JSON.stringify(current)) !== -1;
+        }
 
-    const videoSrc = `https://www.youtube.com/embed/${id}`;
-    const getVideoList = (item) => {
-        return (item.items.map((videoItem, index) => {
-            const { id, snippet } = videoItem;
-            const { title, thumbnails, description } = snippet;
-            
-              return {
-                id: id.videoId,
-                title: title,
-                thumbnail: thumbnails.medium.url,
-                description: description,
-                shortthumbnail: thumbnails.default.url,
-              };
-        }));
-      }
+        return false;
+    }
+    const [isFavorite, setFavorite] = useState(isAlreadyFavorite(state))
+
+    const addToFavorites = () => {
+        let favorites = storage.get(USER_FAVORITE_VIDEOS);
+        if(favorites !== null)
+            favorites.push(state);
+        else
+            favorites = [state];
+
+        storage.set(USER_FAVORITE_VIDEOS, favorites);
+        setFavorite(true);
+    }
 
+    const removeFromFavorites = () => {
+        let favorites = storage.get(USER_FAVORITE_VIDEOS);
+        let current = favorites.findIndex(item => JSON.stringify(item) === JSON.stringify(state));
+        favorites = current === 0 && favorites.length === 1 ? [] : favorites.slice(current, 1);
+        storage.set(USER_FAVORITE_VIDEOS, favorites);
+        setFavorite(false);
+    }
+
+    const selectVideo = (video) => {
+        dispatch(video);
+        setFavorite(isAlreadyFavorite(video));
+    }
+
+    const videoSrc = `https://www.youtube.com/embed/${id}`;
+    
     return(
         <VideoDetailsContainer style={{color: theme.fontcolor, backgroundColor: theme.backgroundcolor }}>
             <div>
                 <VideoPlayer src={videoSrc} allowFullScreen title='Video player'/>
                 <h4>{title}</h4>
                 <p>{description}</p>
+                { !isFavorite ? 
+                <input type="button" value="Agregar a favoritos" onClick={() => addToFavorites()} /> :
+                <input type="button" value="Eliminar de favoritos" onClick={() => removeFromFavorites()} />
+                }
             </div>
             <div>
-                {getVideoList(videoList).map((video, index) => {
-                    const { id, title, description, shortthumbnail} = video;
+                {videoList.map((video, index) => {
+                    const { id, title, description, thumbnail, shortthumbnail} = video;
                     return(
                         id !== undefined &&
-                        <div onClick={(event) => dispatch({ id: id, title: title, description: description })} key={`video-item-container-${id}`}>
+                        <div onClick={() => selectVideo({ id: id, title: title, description: description, thumbnail: thumbnail, shortthumbnail: shortthumbnail })} key={`video-item-container-${id}`}>
```

Comment:
If your arguments has the same name as the variable, you can use shorthands instead to send it to the function.
```suggestion
                        <div onClick={() => selectVideo({ id, title, description, thumbnail, shortthumbnail })} key={`video-item-container-${id}`}>
```